core/src/edu/cornell/gdiac/main/GameController.java

```diff
@@ -345,6 +362,26 @@ use setFrozen() (unimplemented) to freeze the player
       Make it so the player can tap a key to toggle the timer on and off
       Make the timer visually display whether it's enabled
       See my task in input controller */
+
+    if (!input.getMeterPaused()) {
+      metercounter += dt;
+
+      // If moving
+      if (input.getHorizontal() != 0 || input.getVertical() != 0) {
+        metercounter += stationaryconst;
+      }
+
+      // Once meter reaches 100, freeze
+      if (metercounter >= freezeconst) {
+        metercounter = 0;
+        level.getAvatar().setFrozen(true);
```

Comment:
the character should unfreeze after a certain amount of time. You could probably do this by allowing the meter count to get over 100 up to let's say 105 (arbitrary), at which point it completely resets. As in 0-100 is meter rising state, 100-105 is frozen state, and then after which it resets. Play around with this and see what you can do.